sksurv/functions.py

```diff
@@ -75,3 +75,11 @@ def __call__(self, x):
 
     def __repr__(self):
         return "StepFunction(x=%r, y=%r, a=%r, b=%r)" % (self.x, self.y, self.a, self.b)
+
+    def __eq__(self, other):
+        return (
+            all(self.x == other.x)
+            and all(self.y == other.y)
+            and self.a == other.a
+            and self.b == other.b
+        )
```

Comment:
You should add `if isinstance(other, type(self)):` and return `False` otherwise.